packages/ariakit/src/focus-trap/focus-trap-region.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { getAllTabbableIn } from "ariakit-utils/focus";
+import { useForkRef, useWrapElement } from "ariakit-utils/hooks";
+import {
+  createComponent,
+  createElement,
+  createHook,
+} from "ariakit-utils/system";
+import { As, Options, Props } from "ariakit-utils/types";
+import { FocusTrap } from "ariakit/focus-trap";
+
+/**
+ * A component hook that returns props that can be passed to `Role` or any other
+ * Ariakit component to render a focus trap region element.
+ * @see https://ariakit.org/components/focus-trap-region
+ * @example
+ * ```jsx
+ * const props = useFocusTrapRegion();
+ * <Role {...props} />
+ * ```
+ */
+export const useFocusTrapRegion = createHook<FocusTrapRegionOptions>(
+  ({ enabled, ...props }) => {
+    const firstRef = React.useRef<HTMLElement>();
+    const lastRef = React.useRef<HTMLElement>();
+    const container = React.useRef<HTMLDivElement>();
+
+    React.useEffect(() => {
+      if (!container.current) return;
+      const tabbables = getAllTabbableIn(container.current);
+      firstRef.current = tabbables[0];
+      lastRef.current = tabbables[tabbables.length - 1];
+    }, []);
+
+    props = useWrapElement(
+      props,
+      (element) => {
+        const renderFocusTrap = () => {
+          if (!enabled) return null;
+          return (
+            <FocusTrap
+              onFocus={(event) => {
+                if (event.relatedTarget === firstRef.current) {
+                  lastRef.current?.focus();
+                } else {
+                  firstRef.current?.focus();
+                }
+              }}
```

Comment:
Yes I'm doing that on the useEffect above, I thought it would've been better because doing getFirstTabbable and getLastTabbable will call `getAllTabbableIn` twice.

Reply:
My point is that the useEffect will not take lazily mounted elements into account.